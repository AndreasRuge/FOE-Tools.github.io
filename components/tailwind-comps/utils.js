import styleProps from "./props";

/**
 * Checks if object has a specific property.
 * @param {Object} obj
 * @param {String} prop
 */
export const hasOwn = (obj, prop) => Object.prototype.hasOwnProperty.call(obj, prop);

/**
 * Checks if a value is defined
 * @param {*} v
 * @returns {Boolean}
 */
export function isDef(v) {
  return v !== undefined && v !== null;
}

export function useId(size = 3) {
  let uuid = "";
  const dictionary = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
  for (let i = 0; i < size; i++) {
    uuid += dictionary.charAt(Math.floor(Math.random() * dictionary.length));
  }
  return uuid;
}

/**
 * Queries an element from the DOM
 * @param {String} selector Element selector
 * @param {Node} domain HTML element in which to query for element
 * @returns {Node} Node
 */
export const getElement = (selector, domain) => {
  if (!domain) {
    return document.querySelector(selector);
  } else {
    return domain.querySelector(selector);
  }
};

/**
 * @description Wraps and executes both user and internal event handlers for a single event
 * @param {Function} theirHandler Userland event handler
 * @param {*} ourHandler Internal Vue chakra event handler
 */
export const wrapEvent = (theirHandler, ourHandler) => (event) => {
  if (theirHandler) {
    theirHandler(event);
  }

  if (!event.defaultPrevented) {
    return ourHandler(event);
  }
};

/**
 * Converts a kebab-case string into camel case
 * @param {String} string
 */
export function camelize(string) {
  return string.replace(/[.-](\w|$)/g, function (_, x) {
    return x.toUpperCase();
  });
}

/** Filter attrs and return object of chakra props */
export function extractChakraAttrs(attrs) {
  const styleAttrs = {};
  const nativeAttrs = {};

  for (const _prop in attrs) {
    const prop = camelize(_prop);
    if (styleProps[prop]) {
      styleAttrs[prop] = attrs[_prop];
    } else {
      nativeAttrs[_prop] = attrs[_prop];
    }
  }
  return { styleAttrs, nativeAttrs };
}

/**
 * Makes a cache watcher handler for data property.
 * This utility helps prevent unnecessary re-renders
 * for primitives with changes in the $parent $attrs
 * and $listeners objects
 * @param {String} property
 */
export function createWatcher(property) {
  return {
    handler(newVal, oldVal) {
      for (const key in oldVal) {
        if (!hasOwn(newVal, key)) {
          this.$delete(this.$data[property], key);
        }
      }
      for (const key in newVal) {
        this.$set(this.$data[property], key, newVal[key]);
      }
    },
    immediate: true,
  };
}

/**
 * Create mixin for style attributes
 * @param {String} name Component name
 */
export const createStyledAttrsMixin = (name) => ({
  name,
  inheritAttrs: false,
  data() {
    return {
      attrs$: {},
      listeners$: {},
    };
  },
  props: {
    as: {
      type: [String, Object],
      default: "div",
    },
    to: String,
  },
  computed: {
    /** Split style attributes and native attributes */
    splitProps() {
      const $attrs = this.$data.attrs$;
      const styles = Object.assign({}, this.componentStyles || {}, $attrs);

      const { styleAttrs, nativeAttrs } = extractChakraAttrs(styles);
      return {
        styleAttrs,
        nativeAttrs,
      };
    },
    /** Computed attributes object */
    computedAttrs() {
      return {
        ...this.splitProps.nativeAttrs,
      };
    },
    /** Computed listeners object */
    computedListeners() {
      return this.$data.listeners$;
    },
  },
  created() {
    this.$watch("$attrs", createWatcher("attrs$"));
    this.$watch("$listeners", createWatcher("listeners$"));
  },
});

/**
 * Clones a single VNode
 * @param {Vue.VNode} vnodes VNode to be cloned
 * @param {Function} createElement Render function
 * @returns {Vue.VNode} Cloned VNodes
 */
export function cloneVNode(vnode, createElement) {
  const clonedChildren = vnode.children && vnode.children.map((vnode) => cloneVNode(vnode, createElement));
  const cloned = createElement(vnode.tag, vnode.data, clonedChildren);
  cloned.text = vnode.text;
  cloned.isComment = vnode.isComment;
  cloned.componentOptions = vnode.componentOptions;
  cloned.fnOptions = vnode.fnOptions;
  cloned.fnContext = vnode.fnContext;
  cloned.elm = vnode.elm;
  cloned.context = vnode.context;
  cloned.ns = vnode.ns;
  cloned.isStatic = vnode.isStatic;
  cloned.key = vnode.key;
  cloned.data = vnode.data;
  return cloned;
}

/**
 * Clones an array of VNodes
 * @param {Array<Vue.VNode>} vnodes Array of VNodes to be cloned
 * @param {Function} createElement Render function
 * @returns {Array<Vue.VNode>} Cloned VNodes Array
 */
export function cloneVNodes(vnodes, createElement) {
  const clonedVNodes = vnodes.map((vnode) => cloneVNode(vnode, createElement));
  return clonedVNodes;
}

/**
 * Clones VNode with merged data
 * @param {Vue.VNode} vnode VNode
 * @param {Object} data VNode data
 * @param {Function} h Render function
 */
export function cloneVNodeElement(vnode, { props, attrs, children, ...rest }, h) {
  const cloned = cloneVNode(vnode, h);
  return h(
    cloned.componentOptions.Ctor,
    {
      ...cloned.data,
      ...(cloned.componentOptions.listeners || {}),
      props: {
        ...(cloned.data.props || {}),
        ...cloned.componentOptions.propsData,
        ...props,
      },
      attrs: {
        ...(cloned.data.attrs || {}),
        ...attrs,
      },
      ...rest,
    },
    cloned.componentOptions.children || children
  );
}

/**
 * Checks whether a vnode is an async placeholder
 * @param {Vue.VNode} node
 * @returns {Boolean}
 */
function isAsyncPlaceholder(node) {
  return node.isComment && node.asyncFactory;
}

/**
 * Get's the first child component VNode from an array of VNodes
 * @param {Array<Vue.VNode>} children
 * @returns {Vue.VNode}
 */
export function getFirstComponentChild(children) {
  if (Array.isArray(children)) {
    for (let i = 0; i < children.length; i++) {
      const c = children[i];
      if (isDef(c) && (isDef(c.componentOptions) || isAsyncPlaceholder(c))) {
        return c;
      }
    }
  }
}

/**
 * Clears out all undefined or nullish vnodes
 * @param {Array<Vue.VNode>} vnodes Array of VNodes
 * @returns {Array<Vue.VNode>}
 */
export function cleanChildren(vnodes) {
  if (!vnodes) return [];
  return vnodes.filter((vnode) => vnode.tag);
}

/**
 * Extracts native and nonNative event handlers for functional components
 *
 * The returned object returns `native` object contains listeners that
 * MUST be passed to a Vue component in the render function.
 *
 * The returned `nonNative` object contains other native listeners that
 * can be passed to a native HTML element in the render function.
 *
 * @param {Object} context Render function context
 * @param {Object} nonNative Object of VNode `on` event handlers
 * @returns {{ nonNative: Object<String, Function>, native: Object<String, Function>}}
 *
 * @example
 * import Comp from 'comp'
 *
 * const newComp = {
 *  functional: true,
 *  render(h, context) {
 *    const { native, nonNative } = extractListeners(context, componentEvents)
 *    return h(Comp, {
 *      ...context.data
 *      on: nonNative,
 *      nativeOn: native,
 *    }, context.slots ? context.slots().default : context.children)
 *  }
 * }
 */
export const extractListeners = (context, nonNative = {}) => {
  const { listeners } = context;
  const native = {};

  for (const listener in listeners) {
    if (!nonNative[listener]) {
      native[listener] = listeners[listener];
    }
  }

  return { native, nonNative };
};
