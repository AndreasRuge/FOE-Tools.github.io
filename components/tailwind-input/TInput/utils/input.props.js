export const inputProps = {
  as: {
    type: String,
    default: "input",
  },
  _ariaLabel: String,
  _ariaDescribedby: String,
  isFullWidth: {
    type: Boolean,
    default: true,
  },
  focusStyle: {
    type: Boolean,
    default: true,
  },
  isReadOnly: Boolean,
  isDisabled: Boolean,
  isInvalid: Boolean,
  isRequired: Boolean,
  inputId: String,
  rows: Number,
  autoComplete: String,
  value: {
    type: [String, Number],
    default: undefined,
  },
};
