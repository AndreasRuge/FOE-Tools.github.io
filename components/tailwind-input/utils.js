import styleProps from "./props";

/**
 * Checks if object has a specific property.
 * @param {Object} obj
 * @param {String} prop
 */
export const hasOwn = (obj, prop) => Object.prototype.hasOwnProperty.call(obj, prop);

/**
 * Checks if a value is defined
 * @param {*} v
 * @returns {Boolean}
 */
export function isDef(v) {
  return v !== undefined && v !== null;
}

export function useId(size = 3) {
  let uuid = "";
  const dictionary = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
  for (let i = 0; i < size; i++) {
    uuid += dictionary.charAt(Math.floor(Math.random() * dictionary.length));
  }
  return uuid;
}

/**
 * Queries an element from the DOM
 * @param {String} selector Element selector
 * @param {Node} domain HTML element in which to query for element
 * @returns {Node} Node
 */
export const getElement = (selector, domain) => {
  if (!domain) {
    return document.querySelector(selector);
  } else {
    return domain.querySelector(selector);
  }
};

/**
 * @description Wraps and executes both user and internal event handlers for a single event
 * @param {Function} theirHandler Userland event handler
 * @param {*} ourHandler Internal Vue chakra event handler
 */
export const wrapEvent = (theirHandler, ourHandler) => (event) => {
  if (theirHandler) {
    theirHandler(event);
  }

  if (!event.defaultPrevented) {
    return ourHandler(event);
  }
};

/**
 * Converts a kebab-case string into camel case
 * @param {String} string
 */
export function camelize(string) {
  return string.replace(/[.-](\w|$)/g, function (_, x) {
    return x.toUpperCase();
  });
}

/** Filter attrs and return object of chakra props */
export function extractChakraAttrs(attrs) {
  const styleAttrs = {};
  const nativeAttrs = {};

  for (const _prop in attrs) {
    const prop = camelize(_prop);
    if (styleProps[prop]) {
      styleAttrs[prop] = attrs[_prop];
    } else {
      nativeAttrs[_prop] = attrs[_prop];
    }
  }
  return { styleAttrs, nativeAttrs };
}

/**
 * Makes a cache watcher handler for data property.
 * This utility helps prevent unnecessary re-renders
 * for primitives with changes in the $parent $attrs
 * and $listeners objects
 * @param {String} property
 */
export function createWatcher(property) {
  return {
    handler(newVal, oldVal) {
      for (const key in oldVal) {
        if (!hasOwn(newVal, key)) {
          this.$delete(this.$data[property], key);
        }
      }
      for (const key in newVal) {
        this.$set(this.$data[property], key, newVal[key]);
      }
    },
    immediate: true,
  };
}

/**
 * Create mixin for style attributes
 * @param {String} name Component name
 */
export const createStyledAttrsMixin = (name) => ({
  name,
  inheritAttrs: false,
  data() {
    return {
      attrs$: {},
      listeners$: {},
    };
  },
  props: {
    as: {
      type: [String, Object],
      default: "div",
    },
    to: String,
  },
  computed: {
    /** Split style attributes and native attributes */
    splitProps() {
      const $attrs = this.$data.attrs$;
      const styles = Object.assign({}, this.componentStyles || {}, $attrs);

      const { styleAttrs, nativeAttrs } = extractChakraAttrs(styles);
      return {
        styleAttrs,
        nativeAttrs,
      };
    },
    /** Computed attributes object */
    computedAttrs() {
      return {
        ...this.splitProps.nativeAttrs,
      };
    },
    /** Computed listeners object */
    computedListeners() {
      return this.$data.listeners$;
    },
  },
  created() {
    this.$watch("$attrs", createWatcher("attrs$"));
    this.$watch("$listeners", createWatcher("listeners$"));
  },
});
