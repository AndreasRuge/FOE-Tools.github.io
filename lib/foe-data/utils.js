/**
 * Rounds one number to the nearest integer multiple of another
 *
 * @param value {number} The number to round to the nearest integer multiple of another
 * @param factor {number} The number to whose multiples value will be rounded
 * @returns {number} Return the rounded value
 */
function mround(value, factor) {
  return Math.round(value / factor) * factor;
}

/**
 * Generate reward for all of the first 5 places
 *
 * @param firstRankReward {number} Reward of the first place
 * @returns {*[]} Return an array of number
 */
function generateReward(firstRankReward) {
  let result = [firstRankReward];

  result[result.length] = mround(result[result.length - 1] / 2, 5);
  result[result.length] = mround(result[result.length - 1] / 3, 5);
  result[result.length] = mround(result[result.length - 1] / 4, 5);
  result[result.length] = mround(result[result.length - 1] / 5, 5);

  return result;
}

/**
 * Default Ages GB cost (for no age and HMA)
 * @type {*[]} Return array of cost that contains an object with:
 * - cost {number}: level cost
 * - reward {Array}: Array of 5 integer value that represent the reward (in FP) of the first five places
 */
const defaultCost = [
  { cost: 650, reward: generateReward(65) },
  { cost: 667, reward: generateReward(75) },
  { cost: 683, reward: generateReward(85) },
  { cost: 700, reward: generateReward(95) },
  { cost: 718, reward: generateReward(100) },
  { cost: 736, reward: generateReward(110) },
  { cost: 754, reward: generateReward(120) },
  { cost: 773, reward: generateReward(130) },
  { cost: 792, reward: generateReward(140) },
  { cost: 812, reward: generateReward(150) },
  { cost: 833, reward: generateReward(155) },
  { cost: 853, reward: generateReward(165) },
  { cost: 875, reward: generateReward(175) },
  { cost: 897, reward: generateReward(185) },
  { cost: 919, reward: generateReward(195) },
  { cost: 942, reward: generateReward(205) },
  { cost: 965, reward: generateReward(215) },
  { cost: 990, reward: generateReward(225) },
  { cost: 1014, reward: generateReward(235) },
  { cost: 1040, reward: generateReward(245) },
  { cost: 1066, reward: generateReward(255) },
  { cost: 1092, reward: generateReward(265) },
  { cost: 1120, reward: generateReward(275) },
  { cost: 1147, reward: generateReward(285) },
  { cost: 1176, reward: generateReward(300) },
  { cost: 1206, reward: generateReward(310) },
  { cost: 1236, reward: generateReward(320) },
  { cost: 1267, reward: generateReward(330) },
  { cost: 1298, reward: generateReward(340) },
  { cost: 1331, reward: generateReward(350) },
  { cost: 1364, reward: generateReward(365) },
  { cost: 1398, reward: generateReward(375) },
  { cost: 1433, reward: generateReward(385) },
  { cost: 1469, reward: generateReward(395) },
  { cost: 1505, reward: generateReward(405) },
  { cost: 1543, reward: generateReward(420) },
  { cost: 1582, reward: generateReward(430) },
  { cost: 1621, reward: generateReward(440) },
  { cost: 1662, reward: generateReward(450) },
  { cost: 1703, reward: generateReward(465) },
  { cost: 1746, reward: generateReward(475) },
  { cost: 1789, reward: generateReward(495) },
  { cost: 1834, reward: generateReward(500) },
  { cost: 1880, reward: generateReward(510) },
  { cost: 1927, reward: generateReward(520) },
  { cost: 1975, reward: generateReward(535) },
  { cost: 2025, reward: generateReward(545) },
  { cost: 2075, reward: generateReward(555) },
  { cost: 2127, reward: generateReward(570) },
  { cost: 2180, reward: generateReward(580) },
  { cost: 2235, reward: generateReward(590) },
  { cost: 2290, reward: generateReward(605) },
  { cost: 2348, reward: generateReward(615) },
  { cost: 2406, reward: generateReward(630) },
  { cost: 2467, reward: generateReward(640) },
  { cost: 2528, reward: generateReward(650) },
  { cost: 2591, reward: generateReward(665) },
  { cost: 2656, reward: generateReward(675) },
  { cost: 2723, reward: generateReward(690) },
  { cost: 2791, reward: generateReward(700) },
  { cost: 2860, reward: generateReward(715) },
  { cost: 2932, reward: generateReward(725) },
  { cost: 3005, reward: generateReward(735) },
  { cost: 3080, reward: generateReward(750) },
  { cost: 3157, reward: generateReward(760) },
  { cost: 3236, reward: generateReward(775) },
  { cost: 3317, reward: generateReward(785) },
  { cost: 3400, reward: generateReward(800) },
  { cost: 3485, reward: generateReward(810) },
  { cost: 3572, reward: generateReward(825) },
  { cost: 3661, reward: generateReward(835) },
  { cost: 3753, reward: generateReward(850) },
  { cost: 3847, reward: generateReward(860) }
];

module.exports = {
  mround: mround,
  generateReward: generateReward,
  defaultCost: defaultCost
};
