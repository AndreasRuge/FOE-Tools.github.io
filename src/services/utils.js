import { Enum } from "./enumify";

class FormCheck extends Enum {}
FormCheck.initEnum(["VALID", "INVALID", "NO_CHANGE"]);

export default {
  /**
   * Enum of the different results of check
   */
  FormCheck: FormCheck,

  /**
   * Check if a value it's in range. Bound are include.
   *
   * @param value Value to check
   * @param lowerBound Lower bound
   * @param upperBound Upper bound
   * @returns {boolean} Return true if the value it's in range, false otherwise
   */
  inRange(value, lowerBound, upperBound) {
    if (lowerBound > upperBound) {
      return value >= upperBound && value <= lowerBound;
    }
    return value >= lowerBound && value <= upperBound;
  },

  /**
   * Check if an value is valid
   *
   * @param value {number} Value to check
   * @param comparator {Array} Array that contains two elements. The first is a string that corresponding to an operator
   * ("<", "<=", ">", ">=", "==" or "===") or a number. The second is a number.
   * @param currentValue {number} The current value
   * @param type {string} Type of the result value (must be 'int' or 'float')
   * @returns {object} Return an object that contains a "state" attribute for every cases (the type is {#FormCheck}),
   * and for VALID state, it return a "value" attribute.
   */
  checkFormNumeric(value, comparator, currentValue, type = "int") {
    let valid = false;

    if (!isNaN(value)) {
      let resultValue;
      switch (type.toLowerCase()) {
        case "int":
          resultValue = parseInt(value);
          break;
        case "float":
          resultValue = parseFloat(value);
          break;
        default:
          throw new Error(
            'Unexpected type for parameter "type" in checkFormNumeric.'
          );
      }

      if (comparator instanceof Array) {
        if (comparator.length !== 2) {
          throw new Error(
            'Unexpected size array for parameter "comparator" in checkFormNumeric. The comparator array must contain only two numbers.'
          );
        } else if (
          typeof comparator[0] === typeof comparator[1] &&
          typeof comparator[0] === "number"
        ) {
          valid = this.inRange(value, comparator[0], comparator[1]);
        } else if (
          typeof comparator[0] === "string" &&
          typeof comparator[1] === "number"
        ) {
          switch (comparator[0]) {
            case "<":
              valid = resultValue < comparator[1];
              break;
            case "<=":
              valid = resultValue <= comparator[1];
              break;
            case ">":
              valid = resultValue > comparator[1];
              break;
            case ">=":
              valid = resultValue >= comparator[1];
              break;
            case "==":
              valid = resultValue == comparator[1];
              break;
            case "===":
              valid = resultValue === comparator[1];
              break;
            default:
              throw new Error(
                'Unexpected value for parameter "comparator" in checkFormNumeric. Expected a string version of number comparator: "<", "<=", ">", ">=", "==" or "===".'
              );
          }
        } else {
          throw new Error(
            'Unexpected value for parameter "comparator" in checkFormNumeric. The comparator array must contain a number or a string for the fist element and a number for the second.'
          );
        }
      } else {
        throw new Error(
          'Unexpected type for parameter "comparator" in checkFormNumeric. Expect array, found ' +
            typeof comparator +
            "."
        );
      }

      if (valid) {
        if (resultValue !== currentValue) {
          return { state: FormCheck.VALID, value: resultValue };
        }
      }
    }

    if (!valid) {
      return { state: FormCheck.INVALID };
    }

    return { state: FormCheck.NO_CHANGE };
   },

   makeIterator(byLine) {
      let fn = function(arrayList, callback) {
         const BY_LINE = byLine;
         let i         = 0;
         let result    = [];

         for (let gb in arrayList) {
            result[i % BY_LINE] = arrayList[gb];
            if ((i % BY_LINE) === (BY_LINE - 1)) {
               callback(result);
               for (let j = 0; j < result.length; j++) { result[j] = null; }
            }
            i++;
         }

         if ((result[0] !== null) && (result[result.length - 1] === null)) {
            callback(result);
         }
      };
      return fn;
   }
};
