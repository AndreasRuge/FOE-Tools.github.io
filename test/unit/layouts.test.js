import { doTestOf } from "./utils";

describe("Layouts", () => {
  doTestOf("layouts");
});
