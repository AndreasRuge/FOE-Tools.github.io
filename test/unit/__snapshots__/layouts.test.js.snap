// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layouts Default Moving cookies to localStorage 1`] = `
Object {
  "cookieDisclaimerDisplayed": true,
  "currentProfile": "testID",
  "customPromotionMessagesTemplates": Array [],
  "dayNightMode": "auto",
  "dayStart": "08:00",
  "displayTableCard": false,
  "fixedMainMenu": false,
  "gbSelectMode": "datalist",
  "haveReadLocaleInfoAvailable": true,
  "haveReadTipAboutAddInvestor": true,
  "locale": "fr",
  "nightStart": "19:30",
  "profiles": Array [
    Object {
      "id": "testID",
      "name": "Foo Bar",
    },
  ],
  "survey": Array [],
}
`;

exports[`Layouts Default Moving cookies to localStorage 2`] = `
Object {
  "profiles": Object {
    "testID": Object {
      "cf": Object {
        "coins": 123456,
        "cumulativeQuest": 4,
        "fpBy24h": 24,
        "goods": 17,
        "otherRq": 1,
        "secondRq": true,
        "supplies": 654321,
        "suppliesGathered": 80,
      },
      "displayGbName": true,
      "displayTableCard": true,
      "gb": Object {
        "Alcatraz": Object {
          "investorView": Object {
            "customPercentage": false,
            "from": 10,
            "investorPercentageCustom": Array [
              90,
              90,
              90,
              90,
              90,
            ],
            "investorPercentageGlobal": 90,
            "showPlace": Array [
              true,
              true,
              false,
              false,
              false,
            ],
            "takingPlaceInConsideration": 2,
            "to": 60,
          },
          "ownerView": Object {
            "investorParticipation": Array [],
            "investorPercentageCustom": Array [
              92,
              90,
              85,
              80,
              80,
            ],
            "investorPercentageGlobal": 42,
            "level": 21,
            "ownerInvestment": 1,
          },
          "tab": 1,
        },
        "Arctic_Orangery": Object {
          "investorView": Object {
            "customPercentage": false,
            "from": 1,
            "investorPercentageCustom": Array [
              90,
              90,
              90,
              90,
              90,
            ],
            "investorPercentageGlobal": 90,
            "showPlace": Array [
              true,
              false,
              false,
              false,
              false,
            ],
            "takingPlaceInConsideration": 0,
            "to": 10,
          },
          "ownerView": Object {
            "investorParticipation": Array [
              Object {
                "isPotentialSniper": true,
                "value": 1496,
              },
              Object {
                "isPotentialSniper": true,
                "value": 753,
              },
            ],
            "investorPercentageCustom": Array [
              90,
              90,
              90,
              90,
              90,
            ],
            "investorPercentageGlobal": 90,
            "level": 48,
            "ownerInvestment": 0,
          },
          "tab": 0,
        },
        "root": Object {
          "investorView": Object {
            "customPercentage": false,
            "from": 1,
            "investorPercentageCustom": Array [
              90,
              90,
              90,
              90,
              90,
            ],
            "investorPercentageGlobal": 90,
            "showPlace": Array [
              true,
              false,
              false,
              false,
              false,
            ],
            "takingPlaceInConsideration": 0,
            "to": 10,
          },
          "ownerView": Object {
            "investorParticipation": Array [],
            "investorPercentageCustom": Array [
              90,
              90,
              90,
              90,
              90,
            ],
            "investorPercentageGlobal": 90,
            "level": 10,
            "ownerInvestment": 0,
          },
          "tab": 0,
        },
      },
      "gbPrefix": "Hello",
      "gbShowPrefix": true,
      "gbShowSuffix": true,
      "gbSuffix": "World",
      "gbi_tab": 1,
      "promotionMessageList": Array [
        Object {
          "config": Object {
            "displayGbName": true,
            "message": "\${GBN} \${P}",
            "place": "\${PI} (\${PV})",
            "placeSeparator": "  ",
            "prefix": "",
            "reversePlacesOrder": true,
            "showLevel": false,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 0",
          "name": "Default 0",
        },
        Object {
          "config": Object {
            "displayGbName": true,
            "message": "\${GBN} \${FLVL} → \${TLVL} \${P}",
            "place": "P\${PI}(\${PV})",
            "placeSeparator": " ",
            "prefix": "",
            "reversePlacesOrder": false,
            "showLevel": true,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 1",
          "name": "Default 1",
        },
        Object {
          "config": Object {
            "displayGbName": true,
            "message": "\${P} \${GBN} \${FLVL} → \${TLVL}",
            "place": "P\${PI}(\${PV})",
            "placeSeparator": " ",
            "prefix": "",
            "reversePlacesOrder": false,
            "showLevel": true,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 2",
          "name": "Default 2",
        },
        Object {
          "config": Object {
            "displayGbName": true,
            "message": "\${GBN} \${FLVL} → \${TLVL} \${P}",
            "place": "P\${PI}(\${PV})",
            "placeSeparator": " ",
            "prefix": "",
            "reversePlacesOrder": true,
            "showLevel": true,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 3",
          "name": "Default 3",
        },
        Object {
          "config": Object {
            "displayGbName": true,
            "message": "\${P} \${GBN} \${FLVL} → \${TLVL}",
            "place": "P\${PI}(\${PV})",
            "placeSeparator": " ",
            "prefix": "",
            "reversePlacesOrder": true,
            "showLevel": true,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 4",
          "name": "Default 4",
        },
        Object {
          "config": Object {
            "displayGbName": true,
            "message": "\${GBN} \${FLVL} → \${TLVL} \${P}",
            "place": "\${PI}",
            "placeSeparator": " ",
            "prefix": "",
            "reversePlacesOrder": false,
            "showLevel": true,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 5",
          "name": "Default 5",
        },
        Object {
          "config": Object {
            "displayGbName": true,
            "message": "\${GBN} \${FLVL} → \${TLVL} \${P}",
            "place": "\${PI}",
            "placeSeparator": " ",
            "prefix": "",
            "reversePlacesOrder": true,
            "showLevel": true,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 6",
          "name": "Default 6",
        },
        Object {
          "config": Object {
            "displayGbName": true,
            "message": "\${GBN} \${FLVL} → \${TLVL} \${P}",
            "place": "\${PI}",
            "placeSeparator": ",",
            "prefix": "",
            "reversePlacesOrder": false,
            "showLevel": true,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 7",
          "name": "Default 7",
        },
        Object {
          "config": Object {
            "displayGbName": true,
            "message": "\${GBN} \${FLVL} → \${TLVL} \${P}",
            "place": "\${PI}",
            "placeSeparator": ",",
            "prefix": "",
            "reversePlacesOrder": true,
            "showLevel": true,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 8",
          "name": "Default 8",
        },
        Object {
          "config": Object {
            "displayGbName": false,
            "message": "\${GBN} \${FLVL} → \${TLVL} \${P}",
            "place": "\${PI}",
            "placeSeparator": ",",
            "prefix": "",
            "reversePlacesOrder": true,
            "showLevel": false,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 9",
          "name": "Default 9",
        },
        Object {
          "config": Object {
            "displayGbName": true,
            "message": "\${GBN} \${FLVL} → \${TLVL} \${P}",
            "place": "\${PI}",
            "placeSeparator": ",",
            "prefix": "",
            "reversePlacesOrder": true,
            "showLevel": true,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 10",
          "name": "Default 10",
        },
        Object {
          "config": Object {
            "customFields": Object {
              "Your pseudo": Object {
                "key": "Your pseudo",
                "placeholder": "My pseudo",
                "value": "My pseudo",
              },
            },
            "displayGbName": true,
            "message": "\${P}",
            "place": "\${Your pseudo} - \${GBN} \${PI}",
            "placeSeparator": "\${LF}",
            "prefix": "",
            "reversePlacesOrder": false,
            "showLevel": true,
            "suffix": "",
            "useShortGbName": false,
          },
          "id": "Default 11",
          "name": "Default 11",
        },
      ],
      "shortName": false,
      "showLevel": true,
      "showOnlySecuredPlaces": false,
      "showSnipe": false,
      "yourAge": "ColonialAge",
      "yourArcBonus": 90.6,
      "yourCfBoost": 540,
    },
  },
}
`;
